photon-ml/src/main/scala/com/linkedin/photon/ml/FeatureIndexingJob.scala

```diff
@@ -19,40 +19,59 @@ import org.apache.hadoop.conf.Configuration
-  *   just hold them into multiple ones. A heuristically good number could be 1-10, or even as large as 20; but please
+  * The job expects three required arguments and three optional ones:
+  *
+  * input-paths: The input paths of the data
+  * partition-num: The number of partitions to break the storage into. This is merely introduced as an optimization at
+  *   the building stage where we don't necessary have to shuffle all features into the same partition, but rather we
+  *   can hold them in multiple ones. A heuristically good number could be 1-10, or even as large as 20; but please
   *   avoid setting it into an arbitrarily large number. Use just 1 or 2 if you find the indexing job is already fast
   *   enough.
-  * [output_dir]: The output directory
-  * [if_add_intercept] (optional, default=true): whether include INTERCEPT into the map.
-  *
+  * output-path: The output path
+  * date-range: Date range for the input data represented in the form start.date-end.date, e.g. 20150501-20150631
+  * date-range-days-ago: Date range for the input data represented in the form start.daysAgo-end.daysAgo, e.g. 90-1
+  * add-intercept: whether include INTERCEPT in the map (optional, default: true).
+  * data-format: Input data format. (optional, default: TRAINING_EXAMPLE, options:
+  *   [TRAINING_EXAMPLE|RESPONSE_PREDICTION])
+  * feature-shard-id-to-feature-section-keys-map: A map between the feature shard id and it's corresponding feature
+  *   section keys, in the following format: shardId1:sectionKey1,sectionKey2|shardId2:sectionKey2,sectionKey3.
+  * feature-shard-id-to-intercept-map: A map between the feature shard id and a boolean variable that decides whether a
+  *   dummy feature should be added to the corresponding shard in order to learn an intercept, for example, in the
+  *   following format: shardId1:true|shardId2:false. The default is true for all shard ids.
   */
-class FeatureIndexingJob(val sc: SparkContext,
-                         val inputPath: String,
-                         val partitionNum: Int,
-                         val outputDir: String,
-                         val addIntercept: Boolean,
-                         val fieldNames: FieldNames) {
-  private val outputPath: Path = {
-    val path = new Path(outputDir)
+class FeatureIndexingJob(
+    val sc: SparkContext,
+    val inputPaths: Seq[String],
+    val partitionNum: Int,
+    val outputPath: String,
+    val addIntercept: Boolean,
+    val fieldNames: FieldNames,
+    val featureShardIdToFeatureSectionKeysMap: Option[Map[String, Set[String]]] = None) {
+
+  private val logger: PhotonLogger = new PhotonLogger(new Path(outputPath, "_log"), sc)
+
+  /**
+   * Ensures that the output path exists.
+   */
+  private def ensureOutputPath(outputPath: String): Path = {
+    val path = new Path(outputPath)
     val fs = FileSystem.get(new Configuration())
```

Comment:
Use `path.getFileSystem` instead? There is slight different that `path.getFileSystem` would pick the right cluster host while `FileSystem.get(new Configuration())` will use the host where the spark job is residing. It's almost the same but `path.getFileSystem` should be more general (equivalent to `FileSystem.get(new Configuration(), path.toUri())`)